### PR TITLE
feat: ics20 transfer implementation added

### PR DESCRIPTION
**Pull request by gcranju**
### Commit Message

```bash
feat: ics20 transfer implementation added
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)
- [ ] I have added version bump label on PR.

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: contracts/javascore/lib/src/main/java/ibc/icon/interfaces/IICS20Bank.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     * deposit transfers amount of tokenContract from caller to the bank and mints the same amount of tokenContract to the caller.
+     */
+    void deposit(Address tokenContract, BigInteger amount, Address receiver);
+
+    /**
+     * withdraw transfers amount of tokenContract from the bank to receiver and burns the same amount of tokenContract from the bank.
+     */
+    void withdraw(Address tokenContract, BigInteger amount, Address receiver);
+
+
```
Comment (ibrizsabin) [inline contracts/javascore/lib/src/main/java/ibc/icon/interfaces/IICS20Bank.java:40]: do we really need this file now?

**Review by ibrizsabin — COMMENTED** [review]
few concerns , else looks good

File: contracts/javascore/lib/src/main/java/ibc/icon/interfaces/IICS20Bank.java (hunk 1 above)
Comment (gcranju) [reply contracts/javascore/lib/src/main/java/ibc/icon/interfaces/IICS20Bank.java]: missed that, now deleted that file!

File: contracts/javascore/ics20/src/main/java/ibc/ics20/ICS20Transfer.java [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+                }
+            } else {
+                try {
+                    Address tokenContractAddress = getTokenContractAddress(denomOnly);
+                    Context.call(tokenContractAddress, "transfer", receiverAddr, data.amount, data.memo.getBytes());
+                } catch (Exception e) {
+                    ack = ICS20Lib.FAILED_ACKNOWLEDGEMENT_JSON;
+                }
+            }
+        } else {
```
[the remark this answers is not in the document]
Comment (gcranju) [reply contracts/javascore/ics20/src/main/java/ibc/ics20/ICS20Transfer.java]: yeah, makes sense, made the change.

**Comment by izyak** [comment]
LGTM!

**Comment by gcranju** [comment]
Opened in another pr

**Pull request closed without merging**